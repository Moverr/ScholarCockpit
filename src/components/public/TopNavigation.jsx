/* eslint-disable no-unused-expressions */
import React, { Component } from 'react'
import { NavLink } from 'react-router-dom';
import { logTable } from '../../utils/Logger';
import { applicationTitle } from '../../utils/Constants';
import logo from '../../images/logo.png';

import '../../css/nav.css';

const logdata = (
  {
    "section": "TopNavigation",
    "message": "Layout the public menu and navigation ",
    "props": "N/A",

  }
);


function TopNavigation() {
  logTable(logdata);

  return (
    <div className={"row-fluid"}>
      <nav id="bs-navbar" className=" col-md-12 navbar navbar-fixed-top  navbar-expand-lg navbar-light ">
        <ul className="nav navbar-nav">
          <li><img src={logo} alt="Logo" /> <NavLink className={"navbar-brand"} to={'#'} activeClassName="active"> {applicationTitle} </NavLink>
          </li>
        </ul>
        {navigationPanel()} 
      </nav>
    </div>
  )
}


function navigationPanel() {
  return <ul className="nav  navigation_right ">
    <li> <NavLink to={'#'} activeClassName="active"> Product &nbsp; | &nbsp;</NavLink>
    </li>
    <li> <NavLink to={'#'} activeClassName="active"> Pricing &nbsp; | &nbsp;</NavLink>
    </li>
    <li> <NavLink to={'#'} activeClassName="active"> Support &nbsp;</NavLink>
    </li>

  </ul>;
}



export default TopNavigation;

