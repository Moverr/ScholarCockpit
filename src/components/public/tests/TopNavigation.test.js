import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TopNavigation from '../TopNavigation';





let wrapper = shallow(
   
        <TopNavigation />
    
);



describe('Testing   ', () => {


    it('should  render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should have a class banner', () => {
        expect(wrapper).toMatchSnapshot();
    });
})



