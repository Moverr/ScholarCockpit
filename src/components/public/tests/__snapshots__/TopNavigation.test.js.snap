// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing    Should have a class banner 1`] = `
<div
  className="row-fluid"
>
  <nav
    className=" col-md-12 navbar navbar-fixed-top  navbar-expand-lg navbar-light "
    id="bs-navbar"
  >
    <ul
      className="nav navbar-nav"
    >
      <li>
        <img
          alt="Logo"
          src="logo.png"
        />
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navbar-brand"
          to="#"
        >
           
          scholar
           
        </NavLink>
      </li>
    </ul>
    <ul
      className="nav  navigation_right "
    >
      <li>
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="#"
        >
           Product   |  
        </NavLink>
      </li>
      <li>
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="#"
        >
           Pricing   |  
        </NavLink>
      </li>
      <li>
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="#"
        >
           Support  
        </NavLink>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Testing    should  render correctly 1`] = `
<div
  className="row-fluid"
>
  <nav
    className=" col-md-12 navbar navbar-fixed-top  navbar-expand-lg navbar-light "
    id="bs-navbar"
  >
    <ul
      className="nav navbar-nav"
    >
      <li>
        <img
          alt="Logo"
          src="logo.png"
        />
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navbar-brand"
          to="#"
        >
           
          scholar
           
        </NavLink>
      </li>
    </ul>
    <ul
      className="nav  navigation_right "
    >
      <li>
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="#"
        >
           Product   |  
        </NavLink>
      </li>
      <li>
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="#"
        >
           Pricing   |  
        </NavLink>
      </li>
      <li>
         
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="#"
        >
           Support  
        </NavLink>
      </li>
    </ul>
  </nav>
</div>
`;
